docs/documentation/server_admin/topics/admin-console-permissions/fine-grain-v2.adoc

```diff
@@ -178,60 +189,108 @@ assigning only *manage* without *view* will prevent the user from being visible.
 * *Enforce access to*: Defines if the permission should enforce access to all resources of the selected type or to specific resources in a realm.
 * *Policies*: Defines a set of one or more policies that should be evaluated to grant or deny access to the selected resource(s).
 
+After creating the permission, it will automatically take effect when enforcing access to (all) resources and scopes you selected.
+Please, keep that in mind that when creating and updating permissions in production.
+
+===== Defining permissions for viewing realm resources
+
+This feature relies on a partial evaluation mechanism to partially evaluate the permissions that a realm administrator has
+when listing and viewing realm resources. This mechanism will pre-fetch all the permissions set for view-related scopes where the realm administrator
+is referenced either directly or indirectly.
+
+Permissions that grant access to `view` a realm resource of a certain type must use one of the following policies to
+make them available from queries:
+
+* `User`
+* `Group`
+* `Role`
+
+By using any of the policies above, {project_name} can pre-calculate the set of resources that a realm administration can view
+by looking for a direct (if using a user policy) or indirect (if using a role or group policy) reference to the realm administrator.
+Therefore, decorating queries with access controls that will run at the database level. This capability is mainly important to
```

Comment:
Not sure what the subject and verb is here: Therefore, decorating queries with access controls that will run at the database level.

Reply:
The subject is supposed to be the "partial evaluation mechanism" and the verb `decorate` in "decorating queries ...".

Reply:
Okay, I think there is some grammatical problem where I not sure actually what is the verb in the sentence. Is "controls" a verb or a noun. Trying again with the suggestion below.